test/ci-op-configresolver-integration/main.go

```diff
@@ -0,0 +1,143 @@
+	InputConfiguration: api.InputConfiguration{
+		BaseImages: map[string]api.ImageStreamTagReference{
+			"base": {
+				Namespace: "ocp",
+				Name:      "4.2",
+				Tag:       "base",
+			},
+		},
+		BuildRootImage: &api.BuildRootImageConfiguration{
+			ImageStreamTagReference: &api.ImageStreamTagReference{
+				Cluster:   "https://api.ci.openshift.org",
+				Namespace: "openshift",
+				Name:      "release",
+				Tag:       "golang-1.10",
+			},
+		},
+	},
+	Resources: api.ResourceConfiguration{
+		"*": api.ResourceRequirements{
+			Limits: api.ResourceList{
+				"memory": "4Gi",
+			},
+			Requests: api.ResourceList{
+				"cpu":    "100m",
+				"memory": "200Mi",
+			},
+		},
+	},
+	Tests: []api.TestStepConfiguration{
+		{
+			As:       "unit",
+			Commands: "go test ./pkg/...",
+			ContainerTestConfiguration: &api.ContainerTestConfiguration{
+				From: "src",
+			},
+		},
+		{
+			As:       "e2e-aws",
+			Commands: "TEST_SUITE=openshift/conformance/parallel run-tests",
+			OpenshiftInstallerClusterTestConfiguration: &api.OpenshiftInstallerClusterTestConfiguration{
+				ClusterTestConfiguration: api.ClusterTestConfiguration{
+					ClusterProfile: api.ClusterProfileAWS,
+				},
+			},
+		},
+		{
+			As: "e2e-multistage",
+			MultiStageTestConfiguration: &api.MultiStageTestConfiguration{
+				ClusterProfile: api.ClusterProfileAWS,
+				Pre: []api.TestStep{
+					{
+						LiteralTestStep: &api.LiteralTestStep{
+							As:       "ipi-install-rbac",
+							From:     "installer",
+							Commands: "setup-rbac\n",
+							Resources: api.ResourceRequirements{
+								Requests: api.ResourceList{"cpu": "1000m", "mem": "2Gi"},
+							}},
+					}, {
+						LiteralTestStep: &api.LiteralTestStep{
+							As:       "ipi-install-install",
+							From:     "installer",
+							Commands: "openshift-cluster install\n",
+							Resources: api.ResourceRequirements{
+								Requests: api.ResourceList{"cpu": "1000m", "mem": "2Gi"},
+							}},
+					}},
+				Test: []api.TestStep{{
+					LiteralTestStep: &api.LiteralTestStep{
+						As:       "test",
+						From:     "src",
+						Commands: "make custom-e2e",
+						Resources: api.ResourceRequirements{
+							Requests: api.ResourceList{"cpu": "100m", "memory": "200M"},
+						}},
+				}},
+				Post: []api.TestStep{
+					{
+						LiteralTestStep: &api.LiteralTestStep{
+							As:       "ipi-deprovision-must-gather",
+							From:     "installer",
+							Commands: "gather\n",
+							Resources: api.ResourceRequirements{
+								Requests: api.ResourceList{"cpu": "1000m", "mem": "2Gi"},
+							}},
+					},
+					{
+						LiteralTestStep: &api.LiteralTestStep{
+							As:       "ipi-deprovision-deprovision",
+							From:     "installer",
+							Commands: "openshift-cluster destroy\n",
+							Resources: api.ResourceRequirements{
+								Requests: api.ResourceList{"cpu": "1000m", "mem": "2Gi"},
+							}},
+					},
+				},
+			},
+		},
+	},
+}
+
+func main() {
```

Comment:
This seems complex. At a black-box integration level, we will have:

 - registry data
 - config data
 - HTTP requests
 - expected HTTP responses

I almost wonder if we can just write a script that runs the server on the test data and loops through the requests and responses to validate them.